packages/driver/src/cypress/command_queue.ts

```diff
@@ -0,0 +1,400 @@
+import _ from 'lodash'
+import $ from 'jquery'
+import Bluebird from 'bluebird'
+import Debug from 'debug'
+
+import { create as createQueue } from '../util/queue'
+import $dom from '../dom'
+import $utils from './utils'
+import $errUtils from './error_utils'
+
+const debugErrors = Debug('cypress:driver:errors')
+
+interface Command {
+  get(key: string): any
+  get(): any
+  set(key: string, value: any): any
+  set(options: any): any
+  attributes: object
+  finishLogs(): void
+}
+
+const __stackReplacementMarker = (fn, ctx, args) => {
+  return fn.apply(ctx, args)
+}
+
+const commandRunningFailed = (Cypress, state, err) => {
+  // allow for our own custom onFail function
+  if (err.onFail) {
+    err.onFail(err)
+
+    // clean up this onFail callback after it's been called
+    delete err.onFail
+
+    return
+  }
+
+  const current = state('current')
+
+  return Cypress.log({
+    end: true,
+    snapshot: true,
+    error: err,
+    consoleProps () {
+      if (!current) return
+
+      const obj = {}
+      const prev = current.get('prev')
+
+      // if type isnt parent then we know its dual or child
+      // and we can add Applied To if there is a prev command
+      // and it is a parent
+      if (current.get('type') !== 'parent' && prev) {
+        const ret = $dom.isElement(prev.get('subject')) ?
+          $dom.getElements(prev.get('subject'))
+          :
+          prev.get('subject')
+
+        obj['Applied To'] = ret
+
+        return obj
+      }
+
+      return
+    },
+  })
+}
+
+export const create = (state, timeouts, stability, cleanup, fail, isCy) => {
+  const queue = createQueue()
+
+  const get = queue.get.bind(queue)
+  const slice = queue.slice.bind(queue)
+  const at = queue.at.bind(queue)
+  const reset = queue.reset.bind(queue)
+  const clear = queue.clear.bind(queue)
+  const stop = queue.stop.bind(queue)
```

Comment:
i can't help but feel that there's a more concise way to do this using classes, like `class CommandQueue extends Queue`, any reason to not do it that way?

Reply:
I realized all of the binding stuff isn't necessary and was able to clean it up and make it more concise.

Reply:
well, it's not classes, but i'll take it :)